community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPagedFile.java

```diff
@@ -194,27 +196,41 @@ void closeSwapper() throws IOException
     @Override
     public void flushAndForce() throws IOException
     {
+        flushAndForce( IOLimiter.unlimited() );
+    }
+
+    @Override
+    public void flushAndForce( IOLimiter limiter ) throws IOException
+    {
+        if ( limiter == null )
+        {
+            throw new IllegalArgumentException( "IOPSLimiter cannot be null" );
+        }
         try ( MajorFlushEvent flushEvent = tracer.beginFileFlush( swapper ) )
         {
-            flushAndForceInternal( flushEvent.flushEventOpportunity(), false );
+            flushAndForceInternal( flushEvent.flushEventOpportunity(), false, limiter );
             syncDevice();
         }
     }
 
-    public void flushAndForceForClose() throws IOException
+    void flushAndForceForClose() throws IOException
     {
         try ( MajorFlushEvent flushEvent = tracer.beginFileFlush( swapper ) )
         {
-            flushAndForceInternal( flushEvent.flushEventOpportunity(), true );
+            flushAndForceInternal( flushEvent.flushEventOpportunity(), true, IOLimiter.unlimited() );
             syncDevice();
         }
     }
 
-    void flushAndForceInternal( FlushEventOpportunity flushOpportunity, boolean forClosing ) throws IOException
+    void flushAndForceInternal( FlushEventOpportunity flushOpportunity, boolean forClosing, IOLimiter limiter )
+            throws IOException
     {
         pageCache.pauseBackgroundFlushTask();
+        // TODO it'd be awesome if, on Linux, we'd call sync_file_range(2) instead of fsync
```

Comment:
should we have a card instead then, instead of todo that will be deleted eventually?